File: api/domain/Entities/Event.cs
namespace Domain.Entities;

public class Event : BaseEntity
{
    public Event(string description, string observations, DateTime date, Guid dogId, Guid? broodEventId)
    {
        Id = Guid.NewGuid();

        Description = description;
        Date = date;
        DogId = dogId;
        Observations = observations;
        Attachments = new List<Attachment>();
        BroodEventId = broodEventId;

        new EventValidator().Validate(this);
    }

    public string Description { get; protected set; }
    public DateTime Date { get; protected set; }
    public Guid DogId { get; protected set; }
    public string Observations { get; protected set; }
    public ICollection<Guid> AttachmentIds { get; set; }
    public virtual IList<Attachment> Attachments { get; set; }
    public virtual BroodEvent BroodEvent { get; set; }
    public Guid? BroodEventId { get; set; }
}

File: web/src/models/Event.ts
import { Dog } from './Dog';

export interface Attachment {
  id?: string;
  name: string;
}

export interface Event {
  id?: string;
  description: string;
  date: string;
  dogId?: string;
  dog?: Dog;
  base64File?: string;
  attachmentFiles?: File[];
  attachments: Attachment[];
}
